src/test/script_tests.cpp

```diff
@@ -372,114 +372,61 @@ BOOST_AUTO_TEST_CASE(script_build)
                               "P2SH(2-of-3), 1 sig", SCRIPT_VERIFY_P2SH, true
                              ).Num(0).PushSig(keys.key1).Num(0).PushRedeem());
 
-    good.push_back(TestBuilder(CScript() << ToByteVector(keys.pubkey1C) << OP_CHECKSIG,
-                               "P2PK with too much R padding but no DERSIG", 0
-                              ).PushSig(keys.key1, SIGHASH_ALL, 31, 32).EditPush(1, "43021F", "44022000"));
     bad.push_back(TestBuilder(CScript() << ToByteVector(keys.pubkey1C) << OP_CHECKSIG,
-                              "P2PK with too much R padding", SCRIPT_VERIFY_DERSIG
+                              "P2PK with too much R padding", 0
```

Comment:
I don't know why we have some test cases in code and some in JSON; maybe because these are also indirectly testing script construction? Same principles as with the JSON tests: remove the explicit specification of SCRIPT_VERIFY_DERSIG, and then delete any duplicate cases and (unused) cases that we're no longer able to test.